src/setup/rgw.py

```diff
@@ -14,77 +14,220 @@
-        Performs post install operations.
-        """
+    def post_install(cortx_gconf: MappedConf):
+        """Performs post install operations."""
 
+        Log.info('PostInstall phase completed.')
         return 0
 
     @staticmethod
-    def prepare(config_path: str):
-        """
-        Performs prepare phase operations.
-        """
+    def prepare(cortx_gconf: MappedConf):
+        """Perform prepare operations."""
+
+        Log.info('Prepare phase started.')
+
+        try:
+            hostname = cortx_gconf.get(f'node>{Rgw._machine_id}>hostname')
+            host_ip = socket.gethostbyname(hostname)
+            ceph_tmpl_idx = 'ceph_conf_tmpl'
+            # Load template ceph.conf file and updated required field.
+            kv_list = [
+                ('global>fsid', str(uuid.uuid1())),
+                ('global>mon host', host_ip)]
+            Rgw._update_config(ceph_tmpl_idx, ceph_conf_tmpl_url, kv_list)
+
+            # Copy ceph_conf_tmpl_file to /etc/ceph/ dir.
+            if not os.path.exists(ceph_conf_path):
+                os.mkdir(ceph_conf_path)
+            shutil.copyfile(ceph_conf_tmpl_file, ceph_conf_file)
+            Log.info(f'{ceph_conf_tmpl_file} config copied to {ceph_conf_file}.')
+
+            # create admin keyring.
+            cmd = f"sudo ceph-authtool --create-keyring {ceph_admin_keyring_file} \
+                --gen-key -n client.admin --cap mon 'allow *' --cap osd 'allow *'"
+            _, err, rc, = SimpleProcess(cmd).run()
+            if rc != 0:
+                raise RgwSetupError(errno.EINVAL, f'"{cmd}" failed with error {err}.')
+        except Exception as e:
+            raise RgwSetupError(errno.EINVAL, f'Error ocurred while fetching node ip, {e}')
+
+        Log.info('Prepare phase completed.')
 
         return 0
 
     @staticmethod
-    def config(config_path: str):
-        """
-        Performs configurations.
-        """
+    def config(cortx_gconf: MappedConf):
+        """Performs configurations."""
+
+        Log.info('Config phase started.')
 
+        # TODO: Inside [client.rgw.<hostname>] section and
+        # add motr,hax endpoints and rgw client related parameters.
+        Log.info('Config phase completed.')
         return 0
 
     @staticmethod
-    def init(config_path: str):
-        """
-        Perform initialization.
-        """
-
+    def init(cortx_gconf: MappedConf):
+        """Perform initialization."""
+
+        Log.info('Init phase started.')
+        # TODO: Create admin user.
+        # admin user should be created only on one node.
+        # 1. While creating admin user, global lock created in consul kv store.
+        # (rgw_consul_index, cortx>rgw>volatile>rgw_lock, machine_id)
+        # 2. Before creating admin user.
+        #    a. Check for rgw_lock in consul kv store.
+        #    b. Create user only if lock value is None/machine-id.
+
+        rgw_lock = False
+        rgw_lock_key = 'component>rgw>volatile>rgw_lock'
```

Comment:
Try to check logic from legacy S3 server and reuse.

Reply:
Yes I have refereed s3 code, and used the same logic here.
Reference s3 PR: https://github.com/Seagate/cortx-s3server/pull/1623/files